Bumping Stuff to be aligned with the RC Containers .NET 8

microhobby opened the pull request:
@andreriesco take a look

microhobby committed:
monoCsharpForms: update containers base to 6.12.0-rc

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
monoCsharpForms: Change the local attach to launch

This is more easy to the user debug, simple compile and run.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
monoCsharpForms: Bump weston to 4-rc

Also set the torizon.gpuPrefixRC to true in the settings.json file,
so we can use the new GPU prefix in the weston container.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
monoCsharpForms: Add a task to install Nuget packages

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
monoCsharpForms: add a pickString input for the target framework

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
monoCsharpForms: add documentation for specific tasks

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
aspnetBlazor: Update to .net 8.0 template

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

diff --git a/aspnetBlazor/Data/WeatherForecast.cs b/aspnetBlazor/Data/WeatherForecast.cs
index 1c83c40..ce25850 100644
--- a/aspnetBlazor/Data/WeatherForecast.cs
+++ b/aspnetBlazor/Data/WeatherForecast.cs
@@ -2,7 +2,7 @@ namespace __change__.Data;

 public class WeatherForecast
 {
-    public DateTime Date { get; set; }
+    public DateOnly Date { get; set; }

     public int TemperatureC { get; set; }

diff --git a/aspnetBlazor/Data/WeatherForecastService.cs b/aspnetBlazor/Data/WeatherForecastService.cs
index e70d07e..a62a6f2 100644
--- a/aspnetBlazor/Data/WeatherForecastService.cs
+++ b/aspnetBlazor/Data/WeatherForecastService.cs
@@ -7,7 +7,7 @@ public class WeatherForecastService
         "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
     };

-    public Task<WeatherForecast[]> GetForecastAsync(DateTime startDate)
+    public Task<WeatherForecast[]> GetForecastAsync(DateOnly startDate)
     {
         return Task.FromResult(Enumerable.Range(1, 5).Select(index => new WeatherForecast
         {
diff --git a/aspnetBlazor/Pages/FetchData.razor b/aspnetBlazor/Pages/FetchData.razor
index ad6432b..399663b 100644
--- a/aspnetBlazor/Pages/FetchData.razor
+++ b/aspnetBlazor/Pages/FetchData.razor
@@ -1,10 +1,9 @@
 ﻿@page "/fetchdata"
-
-<PageTitle>Weather forecast</PageTitle>
-
 @using __change__.Data
 @inject WeatherForecastService ForecastService

+<PageTitle>Weather forecast</PageTitle>
+
 <h1>Weather forecast</h1>

 <p>This component demonstrates fetching data from a service.</p>
@@ -43,6 +42,6 @@ else

     protected override async Task OnInitializedAsync()
     {
-        forecasts = await ForecastService.GetForecastAsync(DateTime.Now);
+        forecasts = await ForecastService.GetForecastAsync(DateOnly.FromDateTime(DateTime.Now));
     }
 }
diff --git a/aspnetBlazor/Pages/_Host.cshtml b/aspnetBlazor/Pages/_Host.cshtml
index bde06eb..6384ca2 100644
--- a/aspnetBlazor/Pages/_Host.cshtml
+++ b/aspnetBlazor/Pages/_Host.cshtml
@@ -1,8 +1,34 @@
 ﻿@page "/"
+@using Microsoft.AspNetCore.Components.Web
 @namespace __change__.Pages
 @addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
-@{
-    Layout = "_Layout";
-}

-<component type="typeof(App)" render-mode="ServerPrerendered" />
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <base href="~/" />
+    <link rel="stylesheet" href="css/bootstrap/bootstrap.min.css" />
+    <link href="css/site.css" rel="stylesheet" />
+    <link href="Blazor8.styles.css" rel="stylesheet" />
+    <link rel="icon" type="image/png" href="favicon.png"/>
+    <component type="typeof(HeadOutlet)" render-mode="ServerPrerendered" />
+</head>
+<body>
+    <component type="typeof(App)" render-mode="ServerPrerendered" />
+
+    <div id="blazor-error-ui">
+        <environment include="Staging,Production">
+            An error has occurred. This application may no longer respond until reloaded.
+        </environment>
+        <environment include="Development">
+            An unhandled exception has occurred. See browser dev tools for details.
+        </environment>
+        <a href="" class="reload">Reload</a>
+        <a class="dismiss">🗙</a>
+    </div>
+
+    <script src="_framework/blazor.server.js"></script>
+</body>
+</html>
diff --git a/aspnetBlazor/Properties/launchSettings.json b/aspnetBlazor/Properties/launchSettings.json
index c33386a..c9f4806 100644
--- a/aspnetBlazor/Properties/launchSettings.json
+++ b/aspnetBlazor/Properties/launchSettings.json
@@ -3,8 +3,8 @@
     "windowsAuthentication": false,
     "anonymousAuthentication": true,
     "iisExpress": {
-      "applicationUrl": "http://localhost:32394",
-      "sslPort": 44349
+      "applicationUrl": "http://localhost:19325",
+      "sslPort": 44386
     }
   },
   "profiles": {
@@ -12,7 +12,16 @@
       "commandName": "Project",
       "dotnetRunMessages": true,
       "launchBrowser": true,
-      "applicationUrl": "https://localhost:7001;http://localhost:5191",
+      "applicationUrl": "http://localhost:5270",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
+    "https": {
+      "commandName": "Project",
+      "dotnetRunMessages": true,
+      "launchBrowser": true,
+      "applicationUrl": "https://localhost:7230;http://localhost:5270",
       "environmentVariables": {
         "ASPNETCORE_ENVIRONMENT": "Development"
       }
diff --git a/aspnetBlazor/Shared/NavMenu.razor b/aspnetBlazor/Shared/NavMenu.razor
index c3f96b7..b1a5591 100644
--- a/aspnetBlazor/Shared/NavMenu.razor
+++ b/aspnetBlazor/Shared/NavMenu.razor
@@ -7,7 +7,7 @@
     </div>
 </div>

-<div class="@NavMenuCssClass" @onclick="ToggleNavMenu">
+<div class="@NavMenuCssClass nav-scrollable" @onclick="ToggleNavMenu">
     <nav class="flex-column">
         <div class="nav-item px-3">
             <NavLink class="nav-link" href="" Match="NavLinkMatch.All">
diff --git a/aspnetBlazor/Shared/NavMenu.razor.css b/aspnetBlazor/Shared/NavMenu.razor.css
index e681f23..d6beb74 100644
--- a/aspnetBlazor/Shared/NavMenu.razor.css
+++ b/aspnetBlazor/Shared/NavMenu.razor.css
@@ -59,4 +59,10 @@
         /* Never collapse the sidebar for wide screens */
         display: block;
     }
+
+    .nav-scrollable {
+        /* Allow sidebar to scroll for tall menus */
+        height: calc(100vh - 3.5rem);
+        overflow-y: auto;
+    }
 }
diff --git a/aspnetBlazor/Shared/SurveyPrompt.razor b/aspnetBlazor/Shared/SurveyPrompt.razor
index 799bb62..b31a4a6 100644
--- a/aspnetBlazor/Shared/SurveyPrompt.razor
+++ b/aspnetBlazor/Shared/SurveyPrompt.razor
@@ -4,7 +4,7 @@

     <span class="text-nowrap">
         Please take our
-        <a target="_blank" class="font-weight-bold link-dark" href="https://go.microsoft.com/fwlink/?linkid=2149017">brief survey</a>
+        <a target="_blank" class="font-weight-bold link-dark" href="https://go.microsoft.com/fwlink/?linkid=2186158">brief survey</a>
     </span>
     and tell us what you think.
 </div>
diff --git a/aspnetBlazor/__change__.csproj b/aspnetBlazor/__change__.csproj
index b775b09..13d8127 100644
--- a/aspnetBlazor/__change__.csproj
+++ b/aspnetBlazor/__change__.csproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">

   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <Nullable>enable</Nullable>
     <ImplicitUsings>enable</ImplicitUsings>
   </PropertyGroup>
diff --git a/aspnetBlazor/wwwroot/css/open-iconic/README.md b/aspnetBlazor/wwwroot/css/open-iconic/README.md
index 17585a6..64f55be 100644
--- a/aspnetBlazor/wwwroot/css/open-iconic/README.md
+++ b/aspnetBlazor/wwwroot/css/open-iconic/README.md
@@ -8,7 +8,7 @@
 ## What's in Open Iconic?

 * 223 icons designed to be legible down to 8 pixels
-* Super-light SVG files - 61.8 for the entire set
+* Super-light SVG files - 61.8 for the entire set
 * SVG sprite&mdash;the modern replacement for icon fonts
 * Webfont (EOT, OTF, SVG, TTF, WOFF), PNG and WebP formats
 * Webfont stylesheets (including versions for Bootstrap and Foundation) in CSS, LESS, SCSS and Stylus formats
@@ -33,7 +33,7 @@ We like SVGs and we think they're the way to display icons on the web. Since Ope

 Open Iconic also comes in a SVG sprite which allows you to display all the icons in the set with a single request. It's like an icon font, without being a hack.

-Adding an icon from an SVG sprite is a little different than what you're used to, but it's still a piece of cake. *Tip: To make your icons easily style able, we suggest adding a general class to the* `<svg>` *tag and a unique class name for each different icon in the* `<use>` *tag.*
+Adding an icon from an SVG sprite is a little different than what you're used to, but it's still a piece of cake. *Tip: To make your icons easily style able, we suggest adding a general class to the* `<svg>` *tag and a unique class name for each different icon in the* `<use>` *tag.*

 ```
 <svg class="icon">
diff --git a/aspnetBlazor/wwwroot/css/site.css b/aspnetBlazor/wwwroot/css/site.css
index dc4c68b..ed2dca8 100644
--- a/aspnetBlazor/wwwroot/css/site.css
+++ b/aspnetBlazor/wwwroot/css/site.css
@@ -18,6 +18,10 @@ a, .btn-link {
     border-color: #1861ac;
 }

+.btn:focus, .btn:active:focus, .btn-link.nav-link:focus, .form-control:focus, .form-check-input:focus {
+  box-shadow: 0 0 0 0.1rem white, 0 0 0 0.25rem #258cfb;
+}
+
 .content {
     padding-top: 1.1rem;
 }
diff --git a/aspnetBlazor/wwwroot/favicon.ico b/aspnetBlazor/wwwroot/favicon.ico
deleted file mode 100644
index 63e859b..0000000
Binary files a/aspnetBlazor/wwwroot/favicon.ico and /dev/null differ
diff --git a/aspnetBlazor/wwwroot/favicon.png b/aspnetBlazor/wwwroot/favicon.png
new file mode 100644
index 0000000..8422b59
Binary files /dev/null and b/aspnetBlazor/wwwroot/favicon.png differ

microhobby committed:
aspnetBlazor: Add icons and descriptions to local tasks

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
scripts: projectUpdater: Always accept new specific documentation

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
aspnetBlazor: update the update.json file with the files that changed

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
aspnetBlazor: Add cog to run GUI locally and use 4-rc containers

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
aspnetBlazor: Use Debug configuration for publish

In .NET 8 the default configuration for publish is Release, so we need
to specify the configuration to use in the command line.

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
aspnetBlazor: Add mention about the kiosk mode on the docs

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetAvalonia: update to .NET 8.0 and sync with the latest mvvm

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5: Add a new template for Uno using .NET 8 and Uno v5.x

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

microhobby committed:
dotnetUno5FrameBuffer: Add template for Skia.Linux.FrameBuffer

Signed-off-by: Matheus Castello <matheus@castello.eng.br>

andreriesco commented on aspnetBlazor/.vscode/tasks.json:
```diff
@@ -2,7 +2,8 @@
     "tasks": [
         {
-            "label": "build-torizon-local",
+            "label": "build-application",
```
In this tasks would be nice to have `local` in the name also. Like `build-local-application` or `build-application-local`

andreriesco commented on dotnetAvalonia/.doc/README.md:
```diff
@@ -10,4 +10,4 @@ The deployable package is then copied into the running debug container using **s
 
 Finally, remote debugging is performed by attaching to the Visual Studio Debugger (vsdbg) on the running container on the device using a [VSCode feature called Pipe Transport](https://code.visualstudio.com/docs/cpp/pipe-transport). For local debugging, the method used is the one described in the [VSCode documentation for .NET Debugging](https://learn.microsoft.com/en-us/dotnet/core/tutorials/debu ...
```
(https://learn.microsoft.com/en-us/dotnet/core/tutorials/debugging-with-visual-studio-code?pivots=dotnet-8-0)

andreriesco commented on dotnetAvalonia/.vscode/tasks.json:
```diff
@@ -226,7 +230,7 @@
                 "${config:torizon_debug_ssh_port}",
                 "-pr",
-                "${workspaceFolder}/bin/Debug/net6.0/linux-x64/publish",
+                "${workspaceFolder}/bin/Debug/net8.0/linux-x64/publish",
```
Here `${workspaceFolder}/bin/Debug/net8.0/linux-amd64/publish` here, as we use `amd64` to refer to `x86` on the templates

microhobby replied:
I can't change this because is the way that `dotnet publish` works, from the .NET ecosystem the x86-64 or amd64 is x64

andreriesco commented on dotnetAvalonia/Views/MainWindow.axaml:
```diff
@@ -11,6 +11,8 @@
 
     <Design.DataContext>
+        <!-- This only sets the DataContext for the previewer in an IDE,
+             to set the actual DataContext for runtime, set the DataContext property in code (look at App.axaml.cs) -->
```
So, the person should set the `DataContext` property at `App.axaml.cs` ? I am a bit confused with the sentence, but this may just be me as I am not familiarized with `DataContext`

microhobby replied:
This comes from the oficial template, so, not on my charge to change it

andreriesco commented on aspnetBlazor/docker-compose.yml:
```diff
@@ -16,3 +16,91 @@ services:
+      - cog
+
+  cog:
+    image: commontorizon/cog${GPU}:4-rc
```
Can we use the `torizon/cog` one? Seems that for rc there is one [here](https://hub.docker.com/r/torizon/cog-imx8/tags)

microhobby replied:
nope, because the `torizon/cog` does not have the wait for the service script, check https://github.com/commontorizon/Containerfiles/tree/main/cog

andreriesco commented on aspnetBlazor/Properties/launchSettings.json:
```diff
@@ -3,16 +3,25 @@
       "dotnetRunMessages": true,
       "launchBrowser": true,
-      "applicationUrl": "https://localhost:7001;http://localhost:5191",
+      "applicationUrl": "http://localhost:5270",
```
Does this change the default port? Then, we should also change the port on `docker-compose.yml` and on `templates.json`. Also, there is a mention to the 7001 port on the `Program.cs`

microhobby replied:
I do not think that this is really used for something on our workflow, this appears to be something related to Microsoft IIS. But I think that the best is change here the port.

andreriesco commented on monoCsharpForms/.doc/README.md:
```diff
@@ -1 +1,20 @@
+
+## Open MWF Designer
+
+If you don't have access to Visual Studio, for example you are using bare Linux OS instead WSL 2, you can use `Mono Windows Forms Designer` to help you edit a `Form` in graphical mode. To open the designer use the `open-mono-designer` task. This task will use the `Form` class in focus from the VS Code editor as reference to open it on the designer:
```
for example if you are using bare Linux OS instead of WSL 2

andreriesco commented on dotnetUno5/.doc/README.md:
```diff
@@ -0,0 +1,48 @@
+
+### config-hot-reload-address
+
+For the XAML Hot Reload feature to work remotly, running on the Torizon device, the Uno Platform framework needs to know the IP address of the development PC. This task automatically configures the IP address of the development PC in the `<project-name>.Skia.Gtk.csproj`. It's good to run this task always when opening the project.
```
work remotely

andreriesco commented on dotnetUno5/.gitignore:
```diff
@@ -0,0 +1,410 @@
+## Ignore Visual Studio temporary files, build results, and
```
This seems to be for Visual Studio. Is it also valid for VSCode?

microhobby replied:
it's come from the oficial template, so let's leave it here, also is possible to open yet the template using the VS so...

andreriesco commented on dotnetUno5/.vscode/launch.json:
```diff
@@ -0,0 +1,62 @@
+      "stopAtEntry": false
+    },
+    {
+      "name": "Torizon ARMv8",
```
I think Torizon ARMv7 is missing

andreriesco commented on dotnetUno5/.vscode/launch.json:
```diff
@@ -0,0 +1,62 @@
+      },
+      "preLaunchTask": "deploy-torizon-arm64"
+    }
+  ]
```
I think Torizon ARMv7 is missing

microhobby replied:
yeah, this is on purpose. From the telemetry data we have, the use of armv7 is too small to justify new work on it. I'm not saying we're going to abandon it or deprecate it, but we're going to add support now on demand, I think it's fair.

andreriesco replied:
I think is ok for now. Probably in the future we will also move it to "Toradex supported", and so we can add this armv7 one in the future

andreriesco replied:
The armv7 tasks that come from `assets/common.json`  will not be automatically added to the template right?

microhobby replied:
the common tasks will be added, but I believe that they all hide, so I do not see a problem

andreriesco commented on dotnetUno5/.vscode/tasks.json:
```diff
@@ -0,0 +1,160 @@
+    },
+    {
+      "label": "config-hot-reload-address",
+      "detail": "This task configure the __change__.Skia.Gtk project to\nhave remote XAML hot reload. It's good run it everytime you\nopen the project.",
```
This task configures

andreriesco commented on dotnetUno5/.vscode/tasks.json:
```diff
@@ -0,0 +1,160 @@
+          "/property:GenerateFullPaths=true",
+          "/consoleloggerparameters:NoSummary",
+          "-r",
+          "linux-arm64",
```
Here it doesn't need the "-C Debug", like the aspnetBlazor?

andreriesco commented on dotnetUno5/.vscode/tasks.json:
```diff
@@ -0,0 +1,160 @@
+          "color": "terminal.ansiYellow"
+      }
+    },
+  ],
```
Also here the tasks for `arm` 32 bits

microhobby replied:
On purpose.

andreriesco commented on dotnetUno5/Dockerfile.debug:
```diff
@@ -0,0 +1,111 @@
+# Board architecture
+##
+ARG IMAGE_ARCH=
+# For armv7 use:
```
About this comment, does it makes sense remove it from all templates? As this is set automatically when running the extension and there is the validate arch step, the user should not hardcode this

microhobby replied:
yeah I think that make sense

andreriesco commented on dotnetUno5/__change__.Shared/Icons/icon.svg:
Is this blank?

microhobby replied:
yeah, it's come from the oficial template, so

microhobby committed:
aspnetBlazor: launchSettings.json: change port to 7001

As we are using the 7001 in the Program.cs, that is the file
that is used to start the server, let's maintain the consistency

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetUno5/monoCsharpForms: fix typos

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
all: Dockerfiles: Remove comment about armv7

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

andreriesco commented on aspnetBlazor/Dockerfile.debug:
```diff
@@ -3,7 +3,7 @@
 ARG IMAGE_ARCH=
-# For armv7 use:
+
 #ARG IMAGE_ARCH=arm
```
Also this comment, which is a second line of the first one

microhobby committed:
dotnet*: Update the vs code doc link pivots

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetBlazor: tasks: Add -local to the tasks that run on host

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

andreriesco commented on aspnetBlazor/.vscode/launch.json:
```diff
@@ -8,7 +8,7 @@
             "type": "coreclr",
             "request": "launch",
-            "preLaunchTask": "build-torizon-local",
+            "preLaunchTask": "build-application",
```
And here

andreriesco commented on aspnetBlazor/.doc/README.md:
```diff
@@ -2,7 +2,7 @@
 All projects follow the pipeline of tasks described in the [common contributing documentation](https://github.com/toradex/vscode-torizon-templates/blob/bookworm/CONTRIBUTING.md#contributing-templates). However, each project has its own specificities in terms of technologies and methods used to compile, deploy, and debug the code. Therefore, each of them has their own specific tasks in the **tasks ...
 
-This .NET C# console template uses **dotnet publish** CLI command to compile the source code and create a deployable package, with tasks named **publish-torizon-\${architecture}**. However, for local debugging it is not necessary to create a deployable package so just the **dotnet build** task is enough in this case(please note that the **preLaunchTask** of local debug is **build-torizon-local**  ...
+This .NET C# console template uses **dotnet publish** CLI command to compile the source code and create a deployable package, with tasks named **publish-torizon-\${architecture}**. However, for local debugging it is not necessary to create a deployable package so just the **dotnet build** task is enough in this case(please note that the **preLaunchTask** of local debug is **build-application** an ...
```
Please update also here

andreriesco commented on monoCsharpForms/.doc/README.md:
```diff
@@ -11,7 +11,7 @@ the package and their dependencies in the `libs` folder.
 ## Open MWF Designer
 
-If you don't have access to Visual Studio, for example you are using bare Linux OS instead WSL 2, you can use `Mono Windows Forms Designer` to help you edit a `Form` in graphical mode. To open the designer use the `open-mono-designer` task. This task will use the `Form` class in focus from the VS Code editor as reference to open it on the designer:
+If you don't have access to Visual Studio, for example if you are using bare Linux OS instead WSL 2, you can use `Mono Windows Forms Designer` to help you edit a `Form` in graphical mode. To open the designer use the `open-mono-designer` task. This task will use the `Form` class in focus from the VS Code editor as reference to open it on the designer:
```
bare Linux OS instead of WSL 2

microhobby commented:
@andreriesco done, take a look

andreriesco commented:
The only thing that I remembered now is adding the new Uno and Uno FrameBuffer templates to the [README.md](https://github.com/toradex/vscode-torizon-templates/blob/dev/README.md).
Appart from that LGTM

microhobby committed:
README: Update with new Uno 5 templates

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>
